_test/test_utilities_herbert/test_str_to_iarray.m

```diff
@@ -0,0 +1,187 @@
+classdef test_str_to_iarray < TestCase
+    % Test test_str_to_iarray
+    %
+    % str_to_iarray converts a string, character array or cell array of strings
+    % to an integer array
+    
+    methods
+        %--------------------------------------------------------------------------
+        % Single number
+        %--------------------------------------------------------------------------
+        function test_single_number (~)
+            % Test constructor with no arguments
+            iarr = str_to_iarray ('37');
+            assertEqual (iarr, 37)
+        end
+        
+        %--------------------------------------------------------------------------
+        % Matlab style tokens
+        %--------------------------------------------------------------------------
+        function test_range (~)
+            iarr = str_to_iarray ('-3:6');
+            assertEqual (iarr, -3:6)
+        end
+        
+        function test_range_rhs_lt_lhs (~)
+            iarr = str_to_iarray ('6:-3');
+            assertEqual (iarr, 6:-3)
+        end
+        
+        function test_range_with_positive_stride (~)
+            iarr = str_to_iarray ('3:2:8');
+            assertEqual (iarr, 3:2:8)
+        end
+        
+        function test_range_with_negative_stride (~)
+            iarr = str_to_iarray ('3:-2:8');
+            assertEqual (iarr, 3:-2:8)
+        end
+        
+        function test_range_with_positive_stride_rhs_lt_lhs (~)
+            iarr = str_to_iarray ('3:2:-4');
+            assertEqual (iarr, 3:2:-4)
+        end
+        
+        function test_range_with_negative_stride_rhs_gt_lhs (~)
+            iarr = str_to_iarray ('3:-2:8');
+            assertEqual (iarr, 3:-2:8)
```

Comment:
again hides underlying value which is 1x0 double

Reply:
Ditto